src/pages/SignUp/SignUp.js

```diff
@@ -1,6 +1,137 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+// import Input from './Input';
 import './SignUp.scss';
 
-const SignUp = () => {};
+const SignUp = () => {
+  const [inputValues, setInputValues] = useState({
+    name: '',
+    email: '',
+    pw: '',
+    pwCorrect: '',
+  });
+
+  const { name, email, pw, pwCorrect } = inputValues;
+
+  const location = useLocation();
+  const CURRENT_DATA =
+    location.pathname === '/login' ? LOGIN_DATA : SIGNUP_DATA;
+
+  const handleInput = event => {
+    const { name, value } = event.target;
+    setInputValues({ ...inputValues, [name]: value });
+  };
+
+  const checkEmail = email.match(
+    '[a-zA-Z0-9]+[@][a-zA-Z0-9]+[.]+[a-zA-Z]+[.]*[a-zA-Z]*'
+  );
+
+  const conditions = {
+    name: name.length === 0 || name.length >= 2,
+    email: email.length === 0 || checkEmail,
+    pw: pw.length === 0 || pw.length >= 4,
+    pwCorrect: pw === pwCorrect,
+  };
+
+  const signUp = event => {
+    if (name.length >= 2 && pw.length >= 4 && pwCorrect === pw && checkEmail) {
+      event.preventDefault();
+      fetch('http://10.58.52.150:8000/auth/signup', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=utf-8',
+        },
+        body: JSON.stringify({
+          name: `${name}`,
+          email: `${email}`,
+          password: `${pw}`,
+        }),
+      })
+        .then(response => response.json())
+        .then(data => {
+          if (data.accessToken) {
+            localStorage.setItem('token', data.accessToken);
+            location.pathname = '/login';
+          } else {
+            alert('중복된 이메일입니다.');
+            event.preventDefault();
+          }
+        });
+    } else {
+      alert('다시 확인해주세요');
+    }
+  };
+
+  const logIn = event => {
+    if (pw.length >= 4 && checkEmail) {
+      event.preventDefault();
+      fetch('http://10.58.52.150:8000/auth/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=utf-8',
+        },
+        body: JSON.stringify({
+          email: `${email}`,
+          password: `${pw}`,
+        }),
```

Comment:
```suggestion
        body: JSON.stringify({
          email,
          password: pw,
        }),
```
단축속성명을 사용하면 이렇게 할 수도 있습니다. 또한 변수를 가져올때는 굳이 `${}` 로 감싸지 않으셔도 됩니다!